Lib/pathlib/_os.py

```diff
@@ -115,9 +124,15 @@ def copyfile(source, target, follow_symlinks):
             except OSError as err:
                 # Check for ERROR_ACCESS_DENIED
                 if err.winerror != 5 or not _is_dirlink(source):
-                    raise
+                    raise  # Some other error.
             flags |= _winapi.COPY_FILE_DIRECTORY
-        _winapi.CopyFile2(source, target, flags)
+        try:
+            _winapi.CopyFile2(source, target, flags)
+        except OSError as err:
+            # Check for ERROR_INVALID_PARAMETER
+            if err.winerror != 87:
+                raise  # Some other error.
+            raise UnsupportedOperation(err)
```

Comment:
How about this?

```python
        except OSError as err:
            # Handle ERROR_INVALID_PARAMETER due to COPY_FILE_DIRECTORY.
            if err.winerror == 87:
                raise UnsupportedOperation(err)
            raise
```

Also should it be `raise UnsupportedOperation(err) from None` to suppress exception chaining?